plugins/module_utils/role_utils.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+def _sort_rules(rules):
+    for rule in rules:
+        for key in rule:
+            if rule[key]:
+                rule[key].sort()
+    return rules
```

Comment:
@tadeboro I tried with the suggested approach but I had some problems. If I used "one line loop" I was getting generators inside set, which made it impossible to compare. I solved it by implementing a normal for loop that adds a rule one at a time.

Reply:
Generators inside sets? That sounds like a misplaced `)`. I would expect troubles with sets inside a tuple that is then placed in a set itself because the whole tuple becomes unhashable.

This is a bit more polished version of the code that I had in mind:

```
{
    (
        frozenset(r["verbs"] or []),
        frozenset(r["resources"] or []),
        frozenset(r["resource_names"] or [])
    ) for r in rules
}
```

Can you please check again what went wrong? Thanks.

Reply:
Hmm I see, one too many `()` :sweat_smile:.  Wouldn't be better to use `r.get('verbs', []) or []` for cases that key is completely missing?

Reply:
Ugh, yes,  forgot that arguments are not always ansible-supplied and thus some keys can be missing. Great catch. And now you can ignore my comment in review about this ;)